apps/ai-content-generator/src/components/config/content-type-selection/ContentTypeSelection.tsx

```diff
@@ -0,0 +1,74 @@
+import { Dispatch, useState } from 'react';
+import { Checkbox, FormControl, Table } from '@contentful/f36-components';
+import { ContentTypeProps } from 'contentful-management';
+import { ContentTypeText } from '@components/config/configText';
+import { ContentTypeAction, ContentTypeReducer } from '@components/config/contentTypeReducer';
+import { styles } from './ContentTypeSelection.styles';
+
+interface Props {
+  allContentTypes: ContentTypeProps[];
+  selectedContentTypes: string[];
+  dispatch: Dispatch<ContentTypeReducer>;
+}
+
+const ContentTypeSelection = (props: Props) => {
+  const { allContentTypes, selectedContentTypes, dispatch } = props;
+  const [isSelectAllChecked, setIsSelectAllChecked] = useState(false);
+
+  const allContentTypeIds = allContentTypes.map((ct) => ct.sys.id);
+
+  const handleSelectAll = () => {
+    if (isSelectAllChecked) {
+      dispatch({ type: ContentTypeAction.REMOVE_ALL });
+    } else {
+      dispatch({ type: ContentTypeAction.ADD_ALL, value: allContentTypeIds });
+    }
+
+    setIsSelectAllChecked(!isSelectAllChecked);
+  };
+
+  const handleCheckboxChange = (contentType: ContentTypeProps, isChecked: boolean) => {
+    const action = isChecked ? ContentTypeAction.REMOVE : ContentTypeAction.ADD;
+    dispatch({ type: action, value: contentType.sys.id });
+  };
```

Comment:
If we deselect a checkbox when isSelectAllChecked is true, should we set isSelectAllChecked to false?

And vice versa if we select all of the fields by hand, should isSelectedAllChecked become true

Reply:
Good point, added handling for this!